packages/libp2p/src/connection-manager/connection-pruner.ts

```diff
@@ -29,13 +31,22 @@ export class ConnectionPruner {
   private readonly maxConnections: number
   private readonly connectionManager: ConnectionManager
   private readonly peerStore: PeerStore
-  private readonly allow: Multiaddr[]
+  private readonly allow: IpNet[]
   private readonly events: TypedEventTarget<Libp2pEvents>
   private readonly log: Logger
 
   constructor (components: ConnectionPrunerComponents, init: ConnectionPrunerInit = {}) {
     this.maxConnections = init.maxConnections ?? defaultOptions.maxConnections
-    this.allow = init.allow ?? defaultOptions.allow
+    this.allow = (init.allow ?? []).map((ma) => {
+      try {
+        if (!ma.protoNames().includes('ipcidr')) {
+          ma = ma.encapsulate('/ipcidr/32') // Encapsulate with /ipcidr/32 if missing
+        }
+        return convertToIpNet(ma)
+      } catch (error) {
+        throw new Error(`Invalid multiaddr format in allow list: ${ma}`)
+      }
+    })
```

Comment:
This is redundant because `multiaddrToIpNet` adds the `CIDR` tuple if it's missing, and handles IPv6 as well:

```suggestion
    this.allow = (init.allow ?? []).map(ma => multiaddrToIpNet(ma))
```

Reply:
of course can't understand how it happened maybe because too many weeks has passed, and I remember having troubles with with git